app/enterprise/2.1.x/developer-portal/administration/application-registration/azure-oidc-config.md

```diff
@@ -0,0 +1,141 @@
+   the placeholder values for `<your_tenant_id>`, `<your_client_id>`, and `<your_client_secret>`.
+
+   - Get a token from Azure:
+
+     ```bash
+     https -f POST "https://login.microsoftonline.com/<your_tenant_id>/oauth2/v2.0/token" scope=<your_client_id>/.default grant_type=client_credentials -a <your_client_id>:<your_client_secret>
+     ```   
+
+   - Get a token from Kong:
+
+     ```bash
+     https -f POST "https://localhost:8443/httpbin-azure/oauth2/v2.0/token" grant_type=client_credentials -a <your_client_id>:<your_client_secret> --verify NO
+     ```
+
+1. Paste the access token obtained from the previous step into
+   [JWT](https://jwt.io), and copy the value for the
+   [aud (audience)](https://tools.ietf.org/html/rfc7519#section-4.1.3) claim to
+   your clipboard. You will use the `aud` value as your **Reference ID** for the next step.
+
+## Create an Application in Kong
+
+1. Log in to your Developer Portal and create a new application:
+   - Select the **My Apps** menu -> **New Application**.
+   - Enter the **Name** of your Azure application.
+   - Paste the `aud` value generated in JWT in the **Reference ID** field.
+   - (Optional) Enter a **Description**.
+
+   The Create Application form should look similar to this example:
+
+   ![Create Azure Application](/assets/images/docs/dev-portal/azure-app.png)
+
+2. Click **Create**.
+
+3. After you create your application, make sure you activate the Service. In the
+   Services section of the Application Dashboard, click **Activate** on the Service
+   you want to use.
+
+   The view application details page should look similar to this example:
+
+   ![Azure Example Application](/assets/images/docs/dev-portal/azure-app-details.png)
+
+   Because you enabled
+   [Auto-approve](/enterprise/{{page.kong_version}}/developer-portal/administration/application-registration/enable-application-registration##aa)
+   on the associated Application Registration Plugin, an admin won't need to
+   approve the request.
+
+## Test your Authentication Flows with your Azure Application
+
+Follow these instructions to test your client credentials or authorization code
+flows with your Azure AD implementation.
+
+### Test Client Credentials Flow
+
+1. Get a token:
+
+   ```bash
+   https -f POST "https://localhost:8443/httpbin-azure/oauth2/v2.0/token" grant_type=client_credentials -a <your_client_id>:<your_client_secret> --verify NO
+   ```
+
+2. Use the token in an authorization header to retrieve the data:
+
+   ```bash
+   http :8000/httpbin-azure Authorization:'bearer <token_from_above>'
+   ```
+
+   Replace `<token_from_above>` with the bearer token you generated in the previous step.
+
+### Test Authorization Code Flow
+
+In your browser, go to `http://localhost:8000/httpbin-azure`. Replace `localhost`
+with your host information.
+
+You should be guided through a log in process within Azure and then the results
+delivered in your browser.
```

Comment:
```suggestion
If authorization setup was successful, you should be guided through a log in process within Azure.
```
I don't think you need the last part of the sentence here, but that's up to you.

Reply:
stet on Aaron's content